### DISCUSSION TITLE
WARNING (simplemonitor) No alerters defined and no remote logger found

### DISCUSSION THREAD

**Discussion by shiv1all** [jamesoff/simplemonitor#1361] [Q&A] [answered]
Not sure why my alerts block is not being detected here's my simplemonitor.ini file would appreciate it if someone could point me in the right direction, I want to call the Python script which sends a notification to a matrix chat when something goes down. the 'send_alert.py' script is in the same dir as the simplemonitor.ini.

`[monitor]
interval=5
monitors=monitors.ini
alerters=notify

[reporting]
loggers=logfile

[logfile]
type=logfile
filename=simplemonitor.log

[notify]
type=execute
fail_command=./send_alert.py

`

**Comment by jamesoff** [comment, marked as answer]
You just need to move your `alerters=notify` line into the `[reporting]` section :)

```
[monitor]
interval=5
monitors=monitors.ini

[reporting]
loggers=logfile
alerters=notify

[logfile]
type=logfile
filename=simplemonitor.log

[notify]
type=execute
fail_command=./send_alert.py
```

**Reply by shiv1all** [reply]
Thank you so much that works.

**Reply by shiv1all** [reply]
I ran into another issue so here is what I did:
I created the system unit to run the simplemonitor:
---
```
      [Unit]
      Description=SimpleMonitor Service

      [Service]
      ExecStart=/opt/virtualenv/simplemonitor/bin/python /opt/virtualenv/simplemonitor/bin/simplemonitor -f /opt/simplemonitor/simplemonitor.ini
      StandardOutput=append:/var/log/simplemonitor.log
      StandardError=append:/var/log/simplemonitor.log
      User=root
      Group=root

      [Install]
      WantedBy=multi-user.target
```
 ---     
This is my monitors.ini file:
---
```
[monitor]
interval=10 # seconds, to be increased to after testing.
monitors=/opt/simplemonitor/monitors.ini

[reporting]
loggers=logfile
alerters=notify

[logfile]
type=logfile
filename=/var/log/simplemonitor.log

[notify]
type=execute
fail_command=/opt/virtualenv/simplemonitor/bin/python /opt/simplemonitor/send_alert.py "{name}" Down
success_command=/opt/virtualenv/simplemonitor/bin/python /opt/simplemonitor/send_alert.py "{name}" Up
```
---
the monitoring part is running fine it can detect that is the services is down and up in the log file but it does not send the message, when I run the '/opt/virtualenv/simplemonitor/bin/python /opt/simplemonitor/send_alert.py'  it is able to run successfully, I dont know what could be wrong with this one.

**Reply by jamesoff** [reply]
Do you get any output in `/var/log/simplemonitor.log` when it tries to run the commands?

I'm not sure how having systemd append to the same log file that simplemonitor itself is writing to will work - you may want to specify different files in the systemd unit and the simplemonitor configuration.

**Comment by shiv1all** [comment]
Before I only get the `INFO` and `ERROR` output in the `/var/log/simplemonitor.log`.
After adding the --debug tail in the unit file I get the below, don't know how to further drill to get the actual root course of this.
```
2024-03-16 02:43:19    DEBUG (simplemonitor) Running recovery
2024-03-16 02:43:19    DEBUG (simplemonitor) Running alerts
2024-03-16 02:43:19    DEBUG (simplemonitor.alerter-execute) monitor service has failed
2024-03-16 02:43:19    DEBUG (simplemonitor.alerter-execute) not alerting for monitor service: not failed or repeated enough
```

When I turn the service back on:

```
2024-03-16 02:49:19    DEBUG (simplemonitor.alerter-execute) monitor service has recovered
2024-03-16 02:49:19    DEBUG (simplemonitor.alerter-execute) alert for monitor service is SUCCESS
2024-03-16 02:49:19    DEBUG (simplemonitor.alerter-execute) About to execute command: /opt/virtualenv/simplemonitor/bin/python /opt/simplemonitor/send_alert.py "service" Up
2024-03-16 02:49:20    DEBUG (simplemonitor.alerter-execute) Command has finished.
2024-03-16 02:49:20    DEBUG (simplemonitor) Running logs
2024-03-16 02:49:20    DEBUG (simplemonitor) Loop complete
``` 
Alerts still don't come.
When I hit the `/opt/virtualenv/simplemonitor/bin/python /opt/simplemonitor/send_alert.py "service" Up` manually it works fine.
